pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -144,13 +144,19 @@ func (bc *backendPoolTypeNodeIPConfig) ReconcileBackendPools(clusterName string,
 	lbBackendPoolName := getBackendPoolName(clusterName, service)
 	lbBackendPoolID := bc.getBackendPoolID(lbName, bc.getLoadBalancerResourceGroup(), lbBackendPoolName)
 	vmSetName := bc.mapLoadBalancerNameToVMSet(lbName, clusterName)
+	isBackendPoolPreConfigured := bc.isBackendPoolPreConfigured(service)
```

Comment:
Can we just avoid invoking the function? Or returning early.

Reply:
Based on existing logic, if we cannot find a matching backend pool, we need to ignore the `isBackendPoolPreConfigured` option and proceed as it's false. This makes it hard to return early since we anyway have to iterate through all pools.

Reply:
Make sense. Can you add unit test to this change?

Reply:
Sure. Added unit tests and squashed.